### PR TITLE
C++: Fix `cpp/alloca-in-loop` regressions with use-use dataflow

### PR DESCRIPTION
**Pull request by jketema**

### CODE REVIEW

File: cpp/ql/src/Likely Bugs/Memory Management/AllocaInLoop.ql [hunk 1]
```diff
@@ -185,6 +185,16 @@ class LoopWithAlloca extends Stmt {
     not this.conditionReachesWithoutUpdate(var, this.(Loop).getCondition())
   }
 
+  private Expr getExpr(DataFlow::Node node) {
```
Comment (MathiasVP) [inline cpp/ql/src/Likely Bugs/Memory Management/AllocaInLoop.ql:188]: I think this hack is perfectly fine if it gives us back the results we want 👍. However, if we start seeing this hack required in more places we should probably discuss if we need some new API on `DataFlow::Node`

Comment (jketema) [reply cpp/ql/src/Likely Bugs/Memory Management/AllocaInLoop.ql:188]: Agreed.

**Comment by jketema** [comment]
If I'm reading the DCA results correctly, this removes a bunch of bogus results on Samate.

**Comment by MathiasVP** [comment]
> If I'm reading the DCA results correctly, this removes a bunch of bogus results on Samate.

Indeed. It seems like this query was flagging up a bunch of vulnerabilities that happened to be using `alloca` on Samate, but which didn't actually have anything to do with what the query was looking for.

File: cpp/ql/src/Likely Bugs/Memory Management/AllocaInLoop.ql [hunk 2]
```diff
@@ -211,7 +220,11 @@ class LoopWithAlloca extends Stmt {
    */
   private int getAControllingVarInitialValue(Variable var, DataFlow::Node source) {
     source = this.getAPrecedingDef(var) and
-    result = source.asExpr().getValue().toInt()
+    (
+      result = getExpr(source).(Expr).getValue().toInt()
```
Comment (MathiasVP) [inline cpp/ql/src/Likely Bugs/Memory Management/AllocaInLoop.ql:224]: This cast isn't necessary, is it?
```suggestion
      result = getExpr(source).getValue().toInt()
```

**Review by MathiasVP — APPROVED** [review]
LGTM! Just a simple clarifying question.

File: cpp/ql/src/Likely Bugs/Memory Management/AllocaInLoop.ql (hunk 2 above)
Comment (jketema) [reply cpp/ql/src/Likely Bugs/Memory Management/AllocaInLoop.ql:224]: You're right. That seems a leftover of me trying out several things to understand what is going on.

**Pull request merged**